### DISCUSSION TITLE
Installation on QNAP

### DISCUSSION THREAD

**Discussion by Tanarion40** [advplyr/audiobookshelf#3168] [Q&A] [answered]
Hello.
I  have a hard time to make audiobookshelf work on my QNAP.
I can install it, I can launch it, but I can't configure folders. Each time I try to create a new library, it brings me to the wrong place. I have no way to find the path I can access from my file manager.
Any idea ?

Here is my YAML

version: "3.7"
services:
  audiobookshelf:
    image: ghcr.io/advplyr/audiobookshelf:latest
    ports:
      - 13378:80
    volumes:
      - /share/ZFS530_DATA/.qpkg/container-station/data/application/audiobookshelf/audiobooks:/audiobooks
      - /share/ZFS530_DATA/.qpkg/container-station/data/application/audiobookshelf/podcasts:/podcasts
      - /share/ZFS530_DATA/.qpkg/container-station/data/application/audiobookshelf/ebooks:/ebooks
      - /share/ZFS530_DATA/.qpkg/container-station/data/application/audiobookshelf/config:/config
      - /share/ZFS530_DATA/.qpkg/container-station/data/application/audiobookshelf/metadata:/metadata

That are the folders I see when I want to create a library
-.zfs
-bin
-dev
-etc
-home
-lib
-media
-mnt
-opt
-root
-run
-sbin
-srv
-tmp
-usr
-var

**Comment by nichwall** [comment]
Assuming you're using docker, have you set up any mount points?

**Reply by Tanarion40** [reply]
here is my YAML

version: "3.7"
services:
  audiobookshelf:
    image: ghcr.io/advplyr/audiobookshelf:latest
    ports:
      - 13378:80
    volumes:
      - /share/ZFS530_DATA/.qpkg/container-station/data/application/audiobookshelf/audiobooks:/audiobooks
      - /share/ZFS530_DATA/.qpkg/container-station/data/application/audiobookshelf/podcasts:/podcasts
      - /share/ZFS530_DATA/.qpkg/container-station/data/application/audiobookshelf/ebooks:/ebooks
      - /share/ZFS530_DATA/.qpkg/container-station/data/application/audiobookshelf/config:/config
      - /share/ZFS530_DATA/.qpkg/container-station/data/application/audiobookshelf/metadata:/metadata


Folders are created at the right place. It's really when I want to create a library, ABS brings me in a folder from where I can't find the one it created during installation. 

That are the folders I see when I want to create a library
-.zfs
-bin
-dev
-etc
-home
-lib
-media
-mnt
-opt
-root
-run
-sbin
-srv
-tmp
-usr
-var

**Comment by amwmedia** [comment, marked as answer]
I have it running via container station. I can post more details if you're interested

**Reply by nichwall** [reply]
I don't have a QNAP, but sharing more details would be useful (such as screenshots or configuration files depending on the Container Station interface). It is likely a configuration issue

**Reply by Tanarion40** [reply]
Hello.

Yes, I would be interested.
When I installed it it created the folders (ebook, Metadata, etc.) In the
container folder but when I want to create a library I can't find these
folders.

**Reply by amwmedia** [reply]
Here's how I have mine configured. Because this is mounting NAS paths into the default place that ABS stores your config data, updating ABS only requires you to "Recreate Directly". Container Station will pull the new image and apply all the same configurations to the new image.

<img width="800" alt="Screenshot 2024-07-17 at 8 43 21 AM" src="https://github.com/user-attachments/assets/25ec7347-8e90-4199-b62d-23fc54ce5631">

<img width="804" alt="Screenshot 2024-07-17 at 8 43 39 AM" src="https://github.com/user-attachments/assets/9c9fb9b1-a6ab-4ae9-a4bc-bc620ee90540">

<img width="806" alt="Screenshot 2024-07-17 at 8 44 36 AM" src="https://github.com/user-attachments/assets/5505598a-1fb6-49db-bdc1-c87a983e85bf">

<img width="808" alt="Screenshot 2024-07-17 at 8 45 14 AM" src="https://github.com/user-attachments/assets/4544e888-8e6e-4c6a-9d8f-69281c4eb916">

**Reply by Tanarion40** [reply]
Hello. Thank you.

I did this.

All my folders are created. The system added some folder in the metadata one (which mean this part is working at least)

but once in ABS. when i want to create a new library. I'm still stuck in a folder from which I can't find my audiobook folder. I just see these:
-.zfs
-bin
-dev
-etc
-home
-lib
-media
-mnt
-opt
-root
-run
-sbin
-srv
-tmp
-usr
-var

**Reply by Tanarion40** [reply]
Nevermind.  It's working. I just need to create a folder here and it will link to the one on my "audiobooks" folder.
I though that I needed to reach the proper folder :)

Thank you for your help :)

**Reply by alexhk90** [reply]
Thanks, @amwmedia - this helped me get audiobookshelf working using QNAP Container Station. 
In case it's similar for others, the local paths on my NAS (to map as the "Host" for the Storage Mappings / Volumes) all started "/share", e.g. "/share/Audiobooks/OpenAudible/books". Without the "/share" the mounts wouldn't work (showed as blank in audiobookshelf / its container).
